stdlib/public/Concurrency/Actor.cpp

```diff
@@ -434,24 +460,40 @@ extern "C" SWIFT_CC(swift) void _swift_task_enqueueOnExecutor(
     Job *job, HeapObject *executor, const Metadata *executorType,
     const SerialExecutorWitnessTable *wtable);
 
-namespace {
-using SwiftTaskIsCurrentExecutorOptions =
-    OptionSet<swift_task_is_current_executor_flag>;
+/// Check the executor's witness table for specific information about e.g.
+/// being able ignore `checkIsolated` and only call `isIsolatingCurrentContext`.
+static swift_task_is_current_executor_flag
+_getIsolationCheckingOptionsFromExecutorWitnessTable(const SerialExecutorWitnessTable *_wtable) {
+  const WitnessTable* wtable = reinterpret_cast<const WitnessTable*>(_wtable);
+  auto description = wtable->getDescription();
+  if (!description) {
+    return swift_task_is_current_executor_flag::None;
+  }
+
+  if (description->hasNonDefaultSerialExecutorIsIsolatingCurrentContext()) {
+    // The specific executor has implemented `isIsolatingCurrentContext` and
+    // we do not have to call `checkIsolated`.
+    return swift_task_is_current_executor_flag::UseIsIsolatingCurrentContext;
+  }
```

Comment:
This looks good for any conformance compiled against a new standard library, where we can rely on the the bit having been set by IRGen. We don't have to address it in this pull request, but it'll miss the case where the conformance itself was compiled against an older standard library that doesn't have `isIsolatingCurrentContext` defined at all, and the default witness was swapped in at runtime.

Reply:
Yeah, I still have to cover that case -- I think we can test it with a "fake SDK", I've done such before, will follow up on it